ring/client/pool.go

```diff
@@ -173,9 +175,18 @@ func (p *Pool) removeStaleClients() {
 	}
 }
 
-// cleanUnhealthy loops through all servers and deletes any that fails a healthcheck.
+// cleanUnhealthy loops through all servers and deletes any that fails a healthcheck
+// The health checks are executed concurrently with p.cfg.MaxConcurrentHealthChecks.
+// If this is not set (0), then all healthchecks of all clients are executed in parallel.
 func (p *Pool) cleanUnhealthy() {
-	for _, addr := range p.RegisteredAddresses() {
+	addresses := p.RegisteredAddresses()
+
+	maxConcurrent := p.cfg.MaxConcurrentHealthChecks
+	if p.cfg.MaxConcurrentHealthChecks == 0 {
+		maxConcurrent = len(addresses)
```

Comment:
Typically we use a value of 0 to disable a feature. I'm wondering if 0 should be treated as 1. Default to maximum concurrency looks too aggressive to me. Think about the case you run 1k ingesters and you run 1k health checks concurrently.

Reply:
Comment raised in another comment. Having a sane default (like 16) LGTM too.